bson/src/main/org/bson/codecs/pojo/PojoCodecImpl.java

```diff
@@ -266,9 +267,18 @@ private void decodeProperties(final BsonReader reader, final DecoderContext deco
         return codec;
     }
 
+    @SuppressWarnings("unchecked")
     private <S> Codec<S> getCodecFromPropertyRegistry(final PropertyModel<S> propertyModel) {
         try {
-            return propertyCodecRegistry.get(propertyModel.getTypeData());
+            Codec<S> codec = propertyCodecRegistry.get(propertyModel.getTypeData());
+            BsonType representation = propertyModel.getBsonRepresentation();
+            if (representation != null) {
+                if (codec instanceof RepresentationConfigurable) {
+                    return ((RepresentationConfigurable<S>) codec).withRepresentation(representation);
```

Comment:
This cast doesn't seem necessary, and based on your comment above, not necessarily safe.

Reply:
Unless I'm missing something, it's both necessary and safe. The type of `codec` is `Codec` which doesn't necessarily have a `withRepresentation` method, so we need to cast it as `RepresentationConfigurable` before we can call `withRepresentation`. It should be safe because we're checking if `codec` is a `RepresentatoinConfigurable` in the if statement wrapping this line. 

Note that I'm casting `codec`, the caller, not the return value of the call.

Reply:
Whoops, my bad!  I guess this is why they are adding the new pattern matching syntax in Java 15, which will allow you to do:

    if (codec instanceof RepresentationConfigurable rc) {
        rc.withRepresentation(representation);
    }